lib/matplotlib/ticker.py

```diff
@@ -2906,6 +2910,12 @@ def __call__(self):
             return []
 
         if self.ndivs is None:
+            if self.axis.axis_name == 'x':
+                self.ndivs = mpl.rcParams['xtick.minor.ndivs']
+            else:
+                self.ndivs = mpl.rcParams['ytick.minor.ndivs']
```

Comment:
Does this apply to 3D plots? If so, it usually treats z the same as x, while this does the opposite.

Reply:
ah didn’t know. So maybe it could be `if self.axis.axis_name == 'x' or self.axis.axis_name == 'z'` ?

Reply:
Maybe just check 'y' first? Though leave a comment about it.